src/js/domains/Save.js

```diff
@@ -0,0 +1,50 @@
+import { on } from '../utils';
+import { ERROR_MESSAGE, VIDEO } from '../constants';
+
+class Save {
+  static _instance = null;
+
+  static get instance() {
+    if (!Save._instance) {
+      Save._instance = new Save();
+    }
+    return Save._instance;
+  }
+
+  #videos;
+
+  constructor() {
+    this.#videos = this.loadVideos();
+  }
+
+  subscribeEvents(videoItem) {
+    on('.video-item__save-button', '@save', (e) => this.saveVideo(e.detail.videoId), videoItem);
+  }
+
+  saveVideo(videoId) {
+    try {
+      if (this.#videos.length >= VIDEO.MAX_SAVABLE_COUNT) {
+        throw new Error(ERROR_MESSAGE.EXCEED_MAX_SAVABLE_COUNT);
+      }
+
+      localStorage.setItem('videos', JSON.stringify([...this.#videos, { videoId }]));
+      this.#setVideos();
+    } catch (error) {
+      alert(error.message);
+    }
+  }
+
+  getVideos() {
+    return this.#videos;
+  }
+
+  #setVideos() {
+    this.#videos = this.loadVideos();
+  }
+
+  loadVideos() {
+    return localStorage.getItem('videos') ? JSON.parse(localStorage.getItem('videos')) : [];
```

Comment:
```suggestion
    return JSON.parse(localStorage.getItem('videos') ?? '[]');
    // or
    return JSON.parse(localStorage.getItem('videos') ?? [];
```
삼항연산자도 좋지만 이렇게 병합연산자를 활용하는 방법도 괜찮아보이네요 👀 

추가로 LocalStorage의 데이터를 가져오거나 세팅하는 key는 별도의 상수로 관리하는게 유지보수에 좋다고 생각해요

Reply:
와 이런 게 있었군요! 병합 연산자의 존재를 아사님 덕분에 처음 알았습니다 🔥
감사합니다 재빠르게 수정했습니다 🏃🏻‍♀️ [7d5e04b](https://github.com/woowacourse/javascript-youtube-classroom/pull/101/commits/7d5e04be4715216cd83b5c8bd004d723d6602789)

Reply:
로컬 스토리지의 key인 videos를 상수화한다면
`'videos'` -> `LOCAL_STORAGE.VIDEOS` 정도일 것 같은데
과연 상수화의 의미가 있을지 확신이 들지 않습니다 🥲

Reply:
명확히 팀에서 인지한다면 괜찮겠지만 유지보수하는 인원이 늘어나거나 점점 키가 늘어날 경우를 대비해서 드린 코멘트였어요!..
티거님이 그렇게 생각하셨다면 괜찮습니다~ 🙆‍♂️

Reply:
아하 2단계에서 key가 늘어나게 되면 적절한 이름으로 상수화해보겠습니다~! 🙆🏻‍♀️